### PR TITLE
Feature: optimizer.zero_grad() speed up option

### PR DESCRIPTION
**Pull request by ditwoo**
## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)?
- [x] Did you check the code style? `catalyst-make-codestyle && catalyst-check-codestyle` (`pip install -U catalyst-codestyle`).
- [x] Did you make sure to update the docs? We use Google format for all the methods and classes.
- [ ] Did you check the docs with `make check-docs`?
- [x] Did you write any new necessary tests?
- [ ] Did you add your new functionality to the docs?
- [x] Did you update the [CHANGELOG](https://github.com/catalyst-team/catalyst/CHANGELOG.md)?
- [x] **You can use 'Login as guest' to see Teamcity build logs.**

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves an existing feature)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

File: catalyst/core/callbacks/optimizer.py [hunk 1]
```diff
@@ -217,8 +221,17 @@ def on_batch_end(self, runner: IRunner) -> None:
...
+            if not self.use_fast_zero_grad:
+                utils.maybe_recursive_call(self._optimizer, "zero_grad")
+            else:
 
-            utils.maybe_recursive_call(self._optimizer, "zero_grad")
+                def zero_grad(optimizer):
+                    for group in optimizer.param_groups:
+                        for p in group["params"]:
+                            if p.grad is not None:
+                                p.grad = None
```
Comment (ditwoo) [inline catalyst/core/callbacks/optimizer.py]: @Scitator 
how do you think, should i move this function to utils ?

**Review by ditwoo — COMMENTED** [review]
@Scitator 
Could you please suggest some fast (in terms of test speed) test to implement to check that loss function is decreased?
Or should i check gradients? What do you think?

File: catalyst/core/callbacks/optimizer.py (hunk 1 above)
Comment (Scitator) [reply catalyst/core/callbacks/optimizer.py]: yup, could we move to function on top on OptimizerCallback?

**Review by Scitator — COMMENTED** [review]
1 tiny improvement and we are ready to merge :)

File: catalyst/core/callbacks/optimizer.py (hunk 1 above)
Comment (ditwoo) [reply catalyst/core/callbacks/optimizer.py]: thanks, done
